authentication/pipeline/user.py

```diff
@@ -77,6 +85,14 @@ def require_password_and_name_via_email(
     if backend.name != EmailAuth.name or flow != SocialAuthState.FLOW_REGISTER:
         return {}
 
+    if not user:
+        create_user_retval = create_user(
```

Comment:
This is always creating the user so nothing has functionally changed here, I think you need to create the user only if the request has provided a name and password and wrap everything in a transaction to be safe.